### DISCUSSION TITLE
Not of all scheduled taks have been executed

### DISCUSSION THREAD

**Discussion by dharmagetic** [celery/django-celery-beat#553] [Q&A]
* Celery Version: 5.1.2
* Celery-Beat Version: 2.2.1

Hi!
My system has few thouhsands of tasks, scheduled on the same time. Average of one task execution time is 2 seconds. I have a problem that not of all tasks have been executed. 

How can I make that all of my tasks scheduled on the same time should be finished ?

1. Can I use some kind of retrying scheduled Celery tasks ? 
2. Should I use some kind of horizontal scaling e.g. separate tasks by multiple RabbitMQ exchanges ?
3. Should I use some kind of vertical scaling ?

**Comment by sagarc03** [comment]
what happens if you stagger them, for example instead of starting at the everyone starting at the same time, you can group/chuck them and make them start idk 2sec apart.
